File: 03-C# DB Fundamentals/02-DbAdvanced-EntityFramework/15-EXTERNAL FORMAT PROCESSING/ProductsShopSystem/ProductsShopSystem/Data/Models/Category.cs
namespace ProductsShopSystem.Data.Models
{
    using System.Collections.Generic;

    public class Category
    {
        public Category()
        {
            this.Products = new HashSet<CategoryProduct>();
        }

        public int Id { get; set; }

        public string Name { get; set; }

        public HashSet<CategoryProduct> Products { get; set; }
    }
}

File: 07-JavaScript Web/01_ExpressJS/02_Introduction to HTTP and Node.js/ShopStopApp/controllers/category.js
const fs = require('fs');
const Category = require('../models/Category');

module.exports.addGet = (req, res) => {
    res.render('categories/add');
};

module.exports.addPost = async (req, res) => {
    let category = req.body;
    category.creator = req.user._id;
    await Category.create(category);
    res.redirect('/');
};

module.exports.productsByCategory = (req, res) => {
    let categoryName = req.params.category;

    Category.findOne({
            name: categoryName
        })
        .populate('products')
        .then((category) => {
            if (!category) {
                res.sendStatus(404);
                return;
            }

            res.render('categories/products', {
                category
            });
        });
};